### PR TITLE
Mediator for club

### PR DESCRIPTION
**Pull request by andrii52**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by NatalyMinziuk — APPROVED** [review]

**Review by DimaDiD — APPROVED** [review]

File: EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EPlast.BLL.Commands.Club
+{
+    public class UnArchiveCommand : IRequest
+    {
+        public int CLubId { get; set; }
```
Comment (christykobyletska) [inline EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs:11]: interesting spelling: C**L**ubId
is it ok?

File: EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        public UnArchiveCommand(int cLubId)
```
Comment (christykobyletska) [inline EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs:12]: and here:  c**L**ubId

File: EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            {
+                true => x => x.IsActive == isArchive,
+                false => x => x.Name.Contains(clubName) && x.IsActive == isArchive
+            };
+            return expr;
+        }
+
+        private Expression<Func<Club, object>> GetOrder()
+        {
+
```
Comment (christykobyletska) [inline EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs:52]: please remove empty line

File: EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            Expression<Func<Club, object>> expr = x => x.ID;
+            return expr;
+        }
+
+        private Expression<Func<Club, Club>> GetSelector()
+        {
+
```
Comment (christykobyletska) [inline EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs:59]: please remove empty line

File: EPlast/EPlast.BLL/Handlers/ClubHandlers/GetClubDataForReportHandler.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            var clubDto = _mapper.Map<Club, ClubDTO>(club);
+
+            var clubHead = clubAdmins.FirstOrDefault(a => a.AdminType.AdminTypeName == Roles.KurinHead);
+            var head = _mapper.Map<ClubAdministration, ClubReportAdministrationDTO>(clubHead);
+            var clubAdminsDto = _mapper.Map<IEnumerable<ClubAdministration>, IEnumerable<ClubReportAdministrationDTO>>(clubAdmins);
+
+            var clubHistoryFollowersDTO = await _mediator.Send(new GetClubHistoryFollowersQuery(request.ClubId));
+            var clubHistoryMembersDTO = await _mediator.Send(new GetClubHistoryMembersQuery(request.ClubId));
+
+
```
Comment (christykobyletska) [inline EPlast/EPlast.BLL/Handlers/ClubHandlers/GetClubDataForReportHandler.cs:47]: remove one empty line please

File: EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs [hunk 6]
```diff
@@ -45,38 +45,38 @@ public class ClubServiceTests
...
+        //       {
+        //           ClubAdministration = new List<ClubAdministration>(),
+        //           ClubMembers = new List<ClubMembers>()
+        //       });
+        //    _repoWrapper.Setup(r => r.Club.Update(It.IsAny<DataAccessClub.Club>()));
+        //    _repoWrapper.Setup(r => r.SaveAsync());
+
+        //    //Act // Assert
+        //    Assert.ThrowsAsync<InvalidOperationException>(async () => await _clubService.ArchiveAsync(Id));
+        //}
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs:79]: do you need this comments?

File: EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs [hunk 7]
```diff
@@ -1051,42 +911,42 @@ public async Task RemoveAsyncWithoutLogo_SingleNumber_DeletesClubWithotLogo()
...
+        //        .Setup(r => r.ClubAdministration.GetAllAsync(It.IsAny<Expression<Func<DataAccessClub.ClubAdministration, bool>>>(), null))
+        //        .ReturnsAsync(new List<DataAccessClub.ClubAdministration>());
+
+        //    // Act
+        //    var result = await _clubService.GetClubAdministrations(Id);
+
+        //    // Assert
+        //    Assert.NotNull(result);
+        //    Assert.IsInstanceOf<DataAccessClub.ClubAdministration[]>(result);
+        //}
```
Comment (christykobyletska) [inline EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs:949]: and these

File: EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs (hunk 1 above)
Comment (andrii52) [reply EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs]: It's definitely not ok, fixed that

File: EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs (hunk 3 above)
Comment (andrii52) [reply EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs]: Removed, thx

File: EPlast/EPlast.BLL/Handlers/ClubHandlers/GetClubDataForReportHandler.cs (hunk 5 above)
Comment (andrii52) [reply EPlast/EPlast.BLL/Handlers/ClubHandlers/GetClubDataForReportHandler.cs]: Done

File: EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs (hunk 7 above)
Comment (andrii52) [reply EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs]: Thanks, done

File: EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs (hunk 6 above)
Comment (andrii52) [reply EPlast/EPlast.Tests/Services/Club/ClubServiceTests.cs]: No, just forgot to delete, thx

File: EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs (hunk 4 above)
Comment (andrii52) [reply EPlast/EPlast.BLL/Handlers/ClubHandlers/GetAllClubsByPageAndIsArchiveHandler.cs]: Done

File: EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs (hunk 2 above)
Comment (andrii52) [reply EPlast/EPlast.BLL/Commands/Club/UnArchiveCommand.cs:12]: Fixed

**Review by christykobyletska — APPROVED** [review]

**Pull request merged**